docs/source/math/function.rst

```diff
@@ -10,6 +10,56 @@ documentation and the Cherab function tutorials.
 Cherab previously provided vector functions which were not present in Raysect.
 New codes should prefer the Raysect vector functions, but the old aliases are preserved for backwards compatibility.
 
+The Function6D framework in Cherab aims to provide a framework for building six-dimensional distribution functions.
+The relation of Function6D to distribution functions makes it domain specific, and so it was included in Cherab's math module.
```

Comment:
I don't think you need to justify including Function6D here: the first line you've added is sufficient to motivate the framework, and the third line is sufficient to highlight the similarity to Raysect.

Reply:
I've replaced the nested for cycles with the itertools.product.

Reply:
I removed the 2nd line